Classes/Hooks/Interfaces/TimeSpan.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Hooks\Interfaces;
+
+/**
+ * Use this interface for hooks concerning the date span.
+ *
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+interface TimeSpan extends Hook
+{
+    /**
+     * Modifies the date span string.
+     *
+     * The date format for the date parts are configured in TypoScript (`dateFormatYMD` etc).
+     * This allows modifying the assembly of start and end date to the date span.
+     * E.g. for Hungarian: '01.-03.01.2019' -> '2019.01.01.-03.'.
+     *
+     * @param string $dateSpan the date span produced by `AbstractTimeSpan::getDate()`
+     * @param string $beginDate the formatted begin date part (`dateFormatYMD`)
+     * @param string $dash the glue used by `AbstractTimeSpan::getDate()` (may be HTML encoded)
+     * @param string $endDate the formatted end date part (`dateFormatYMD`)
+     *
+     * @return string the modified date span to use
+     */
+    public function modifyDateSpan(string $dateSpan, string $beginDate, string $dash, string $endDate): string;
```

Comment:
Wouldn't it be more flexible to pass the begin and end date as timestamps?

Reply:
It'd need to get the configured format strings into the hooked-in method, also. That's why I chose to use the pre-formatted strings. I'll take a look into a way of passing all that without dozens of parameters.

Reply:
I could pass `$this` (the instance of `\Tx_Seminars_OldModel_AbstractTimeSpan`) to the methods. One could use `::getBeginDateAsTimestamp()` as well as `getConfValueString('dateFormatYMD')` to get the required values from the caller. I would keep `$dateSpan` (for smaller adjustments) and `$dash` (to indicated HTML encoded or not) also. OK?

Reply:
@mk-mxp Sounds good.